generators/server/templates/src/test/java/package/web/rest/KafkaResourceIT.java.ejs

```diff
@@ -18,68 +18,44 @@
-import <%=packageName%>.RedisTestContainerExtension;
-<%_ } _%>
-<%_ if (authenticationType === 'uaa') { _%>
-import <%=packageName%>.config.SecurityBeanOverrideConfiguration;
-<%_ } _%>
-<%_ if (authenticationType === 'oauth2') { _%>
-import <%=packageName%>.config.TestSecurityConfiguration;
-<%_ } _%>
-import <%= packageName %>.service.<%= upperFirstCamelCase(baseName) %>KafkaProducer;
-import <%= packageName %>.service.<%= upperFirstCamelCase(baseName) %>KafkaConsumer;
-import org.apache.kafka.common.Metric;
-import org.apache.kafka.common.MetricName;
-import org.assertj.core.api.Assertions;
+import <%= packageName %>.config.KafkaProperties;
+import <%= packageName %>.web.rest.<%= upperFirstCamelCase(baseName) %>KafkaResource.SafeKafkaConsumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-<%_ if (cacheProvider === 'redis') { _%>
-import org.junit.jupiter.api.extension.ExtendWith;
-<%_ } _%>
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.test.annotation.DirtiesContext;
 import org.springframework.test.web.servlet.MockMvc;
 import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.util.ReflectionUtils;
 import org.testcontainers.containers.KafkaContainer;
 
+import java.lang.reflect.Field;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
 import java.util.Map;
+import java.util.Set;
 
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
-<%_ if (authenticationType === 'oauth2') { _%>
-@SpringBootTest(classes = {<%= mainClass %>.class, TestSecurityConfiguration.class})
-<%_ } else if (authenticationType === 'uaa' && applicationType !== 'uaa') { _%>
-@SpringBootTest(classes = {SecurityBeanOverrideConfiguration.class, <%= mainClass %>.class})
-<%_ } else { _%>
-@SpringBootTest(classes = <%= mainClass %>.class)
-<%_ } _%>
-@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_CLASS)
-<%_ if (cacheProvider === 'redis') { _%>
-@ExtendWith(RedisTestContainerExtension.class)
-<%_ } _%>
-public class <%= upperFirstCamelCase(baseName) %>KafkaResourceIT {
-
-    private MockMvc restMockMvc;
+class <%= upperFirstCamelCase(baseName) %>KafkaResourceIT {
```

Comment:
- can we add `public class` ?
- then, is it a real IT test ? Otherwise, maybe Test instead

Reply:
Why should it be public ? It was a proposition by IDEA to remove the public

Reply:
to be consistent with our other class, but it's fine for me here

Reply:
About whether it's a UT or IT, that's a good question. It doesn't spawn the Spring context but it spawns a real Kafka with testcontainer.

Reply:
I would say to keep it package private as it does not need to be public (and we could change the others for consistency if they don't need to be public).
For me that's more an integration test than a unit one.